### PR TITLE
[@task] remove overloaded constructor

### PR DESCRIPTION
**Pull request by taylor-vann**
Currently, we overload the constructor of `task`.

However, Typescript is smart enough and the overloads are similar enough that typescript can do the work of the three overloads itself.

As in this constructor:
```ts
  constructor(
    host: ReactiveControllerHost,
    task: TaskFunction<T, R> | TaskConfig<T, R>,
    args?: ArgsFunction<T>
  ) {
```
Is categorically the same as these three constructors:
```ts
  constructor(
    host: ReactiveControllerHost,
    task: TaskFunction<T, R>,
    args?: ArgsFunction<T>
  );
  constructor(host: ReactiveControllerHost, task: TaskConfig<T, R>);
  constructor(
    host: ReactiveControllerHost,
    task: TaskFunction<T, R> | TaskConfig<T, R>,
    args?: ArgsFunction<T>
  ) {
```

The argument types between overloads are not different enough for overloading a constructor. But in the future, if completely different arg names / types are needed we can add more constructors! <3

### CODE REVIEW

File: packages/labs/task/src/task.ts [hunk 1]
```diff
@@ -125,12 +125,6 @@ export class Task<T extends [...unknown[]] = any, R = any> {
   private _resolveTaskComplete!: (value: R) => void;
   private _rejectTaskComplete!: (e: unknown) => void;
 
-  constructor(
```
Comment (sorvell) [inline packages/labs/task/src/task.ts:128]: Is the idea that the overrides don't actually do anything so we're deleting them? Maybe update the PR comment to make that more clear.

Comment (taylor-vann) [reply packages/labs/task/src/task.ts:128]: yes exactly. The overloads are similar enough that typescript can do the work of the three overloads itself.

But in the future if there was a more different constructor, like different args param names, we'd need an other constructor.

also sounds good!

Comment (taylor-vann) [reply packages/labs/task/src/task.ts:128]: done!

**Review by taylor-vann — COMMENTED** [review]
addressed review changes :)

File: .changeset/wild-ligers-cross.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@lit-labs/task': patch
```
Comment (sorvell) [inline .changeset/wild-ligers-cross.md:2]: If this is actually a no-op change, it shouldn't need a release, right?

Comment (taylor-vann) [reply .changeset/wild-ligers-cross.md:2]: yeup, you're correct. sounds good!

File: .changeset/wild-ligers-cross.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+---
+
+remove constructor overloads
```
[the remark this answers is not in the document]
Comment (taylor-vann) [reply .changeset/wild-ligers-cross.md:5]: er ... empty changesets are empty. my bad, will be more explicit in the future though!

**Review by taylor-vann — COMMENTED** [review]
feedback addressed!

File: .changeset/wild-ligers-cross.md (hunk 2 above)
Comment (taylor-vann) [reply .changeset/wild-ligers-cross.md]: i think we commented in the same cache frame lol

**Comment by taylor-vann** [comment]
@AndrewJakubowicz @sorvell 
Added a more explicit patch changeset :)
Should be good to go, ready for another review

**Review by justinfagnani — CHANGES_REQUESTED** [review]
This removes information from the constructor types that points to the valid usage: that it's not valid to pass both a TaskConfig and an args parameter.

**Comment by taylor-vann** [comment]
@justinfagnani If you could clarify a little bit, i don't understand. This change is categorically the same as the overloaded constructor

**Comment by taylor-vann** [comment]
Ah if i understand correctly then you want an overloaded constructor, despite the types being the same it's the use case that's different. If that's the case I'll close this PR and move on.

**Comment by justinfagnani** [comment]
Yeah, one of the things with TypeScript overloads is that the non-overload signature _must_ be a superset of all of the overloads, but it may not express the constraints between the parameters that an overload can.

Class example is:
```ts
function foo(a: X, b: Y): void;
function foo(a: Z, b: W): void;
function foo(a: X | Z, b: Y | W): void {...}
```

The type `function foo(a: X | Z, b: Y | W): void` doesn't express that when `a` is an `X`, `b` must be a `Y`.

**Pull request closed without merging**